src/OrbitGl/CaptureViewElementTest.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include "CaptureViewElementTester.h"
+
+namespace orbit_gl {
+
+class UnitTestCaptureViewElement : public CaptureViewElement {
+ public:
+  explicit UnitTestCaptureViewElement(CaptureViewElement* parent, TimeGraph* time_graph,
+                                      Viewport* viewport, TimeGraphLayout* layout,
+                                      int child_count = 0)
+      : CaptureViewElement(parent, time_graph, viewport, layout) {
+    for (int i = 0; i < child_count; ++i) {
```

Comment:
Maybe I'm missing something, but why you used this fake `child_count ` instead of some method inside the classes themselves.

Reply:
Which method are you thinking of? The main point of this is that I want to test functionality implemented in `CaptureViewElement`, so I need to create a class inheriting from `CaptureViewElement` to run those tests because the class  itself is pure virtual. I don't want to use any of the existing classes though because those could introduce their own bugs.

I could have just defined a hard-coded number of children for this test, but I suspect we could extend this class for additional tests, so I kept it a bit flexible.